### PR TITLE
added Fibonacci number generator procedures

### PR DESCRIPTION
**Pull request by M-CS-ME**
This file contains 3 procedures for generating Fibonacci numbers.

- the first one is recursive,
- the second one is iterative-recursive (tail-call),
- the last is iterative using a while loop.

### CODE REVIEW

File: src/math/fib.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""
+Program for calculating the nth fibonacci number
```
Comment (Panquesito7) [inline src/math/fib.jl:2]: Please add a source or Wikipedia link for algorithm explanation.
You can also add a link to your GitHub profile at the bottom like

```markdown
Contributed By:- [Your name](Link to your GitHub profile)
```

File: src/math/fib.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+"""
+
+#Recursive Procedure
+function fib_rec(n)
```
Comment (Panquesito7) [inline src/math/fib.jl:6]: Please add detailed function documentation (including parameters).

File: src/math/fib.jl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	if n < 2
+		return n
+	end
+	return fib_rec(n-1) + fib_rec(n-2)
+end
+
+#Iterative Recursive Procedure
+function fib_rec_iter(n)
```
Comment (Panquesito7) [inline src/math/fib.jl:14]: Same here (missing documentation).

File: src/math/fib.jl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		if c == 0
+			return a
+		end
+		return iter(b, a+b, c-1)
+	end
+	return iter(0, 1, n)
+end
+
+#Iterative Procedure
+function fib_iter(n)
```
Comment (Panquesito7) [inline src/math/fib.jl:25]: Same here (missing documentation).

**Comment by Azzaare** [comment]
Thanks @M-CS-ME for the addition!

Could you add some tests in `test/math.jl`?

It is something we want to ask in the guideline for contribution, but unfortunately, we haven't written things yet.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by Azzaare** [comment]
As we don't have answer from @M-CS-ME at the moment, this PR becomes low priority.

Documentation and tests are required. Will make it draft for now.

**Comment by uncomfyhalomacro** [comment]
@M-CS-ME Hello. How is your progress of this PR? Sorry for the ping.

**Comment by Azzaare** [comment]
This is almost two months old. I think we should close the PR (the branch is still alive anyway, and we can always resurrect the PR).

What do you think @uncomfyhalomacro @Panquesito7 ?

**Pull request closed without merging**